### PR TITLE
[Delivers #96904782] Custom email subjects for NCR

### PR DESCRIPTION
**Pull request by cmc333333**
Leverages Rails' internationalization/localization infrastructure to pull out the email subjects. This allows interpolation with fields within the `client_data` object.

### CODE REVIEW

File: app/mailers/communicart_mailer.rb [hunk 1]
```diff
@@ -113,9 +113,24 @@ def send_proposal_email(proposal: nil, to_email: nil, from_email: nil, template_
...
+    params[:public_identifier] = proposal.public_identifier
+    proposal.requester.as_json.each { |k, v| params["requester_" + k] = v }
+    key = :proposal
+    if proposal.client_data
+      key = proposal.client_data.class.name.underscore
+      params.merge!(proposal.client_data.as_json)
+    end
+    params[:scope] = [:mail, :subject]
+    params[:default] = :proposal
+    I18n.t key, params.symbolize_keys
```
Comment (afeld) [inline app/mailers/communicart_mailer.rb]: A little tough to understand what is happening in this method...maybe worth some inline comments, or splitting into smaller ones?

File: spec/mailers/communicart_mailer_spec.rb [hunk 2]
```diff
@@ -225,4 +225,18 @@ def sender_names(mail)
...
+    it 'defaults when no client_data is present' do
+      proposal = FactoryGirl.create(:proposal)
+      mail = CommunicartMailer.proposal_created_confirmation(proposal)
+      expect(mail.subject).to eq("Request " + proposal.public_id)
+    end
+
+    it 'includes custom text for ncr work orders' do
+      wo = FactoryGirl.create(:ncr_work_order)
+      mail = CommunicartMailer.proposal_created_confirmation(wo.proposal)
+      expect(mail.subject).to eq("Request #{wo.public_identifier}, #{wo.org_code}, #{wo.building_number}, from #{wo.requester.email_address}")
```
Comment (afeld) [inline spec/mailers/communicart_mailer_spec.rb]: Might want to consider setting explicit values when creating the instance and checking for explicit values here.

File: app/mailers/communicart_mailer.rb (hunk 1 above)
Comment (cmc333333) [inline app/mailers/communicart_mailer.rb]: Sure thing

File: spec/mailers/communicart_mailer_spec.rb (hunk 2 above)
Comment (cmc333333) [inline spec/mailers/communicart_mailer_spec.rb]: Good idea

**Comment by cmc333333** [comment]
Updated!

**Pull request merged**